sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubClientFactory.cs

```diff
@@ -48,14 +48,14 @@ internal EventHubProducerClient GetEventHubProducerClient(string eventHubName, s
             eventHubName = _nameResolver.ResolveWholeString(eventHubName);
             connection = _nameResolver.ResolveWholeString(connection);
 
-            return _producerCache.GetOrAdd(eventHubName, key =>
+            if (!string.IsNullOrWhiteSpace(connection))
             {
-                if (!string.IsNullOrWhiteSpace(connection))
-                {
-                    var info = ResolveConnectionInformation(connection);
+                var info = ResolveConnectionInformation(connection);
 
-                    if (info.FullyQualifiedEndpoint != null &&
-                        info.TokenCredential != null)
+                if (info.FullyQualifiedEndpoint != null &&
+                    info.TokenCredential != null)
+                {
+                    return _producerCache.GetOrAdd(info.FullyQualifiedEndpoint + eventHubName, key =>
```

Comment:
We should probably do something like:
$"{fullyQualifiedNamespace}/{entityPath}"

Reply:
Fixed